app/features/build_runner/build_runner.rb

```diff
@@ -139,6 +188,15 @@ def reset_repo_state
     def post_run_action
       logger.debug("Finished running #{self.project.project_name} for #{self.sha}")
       self.reset_repo_state
+
+      self.unset_build_specific_environment_variables
```

Comment:
let's make this the first action for `post_run_action`, in case we have an exception. Also we might want to: #todo: make `post_run_action` run no matter what (like an ensure block). Or provide `post_run_ensure`... thoughts?

Reply:
Oh yes, I didn't know we don't guarantee running it yet, since I thought we always want to run `reset_repo_state`. I'll create an issue for it 👍